### PR TITLE
Feature/313 get user projects

### PR DESCRIPTION
**Pull request by waltersajtos**
## Description

For the project profile page we are going to need an endpoint to fetch the projects that belong to the user

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [ ] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Create a project on a user
2. Make a request to {dexAPI}/user/projects
3. The API should return the user's projects

## Link to issue
#313 

Closes: #313

### CODE REVIEW

File: API/Controllers/UserController.cs
Comment (Brend-Smits) [inline API/Controllers/UserController.cs:194]: Formatting is off?

File: API/Controllers/UserController.cs
Comment (Brend-Smits) [inline API/Controllers/UserController.cs:168]: I think you can just use the helper method GetContextUser

File: Repositories/ProjectRepository.cs
Comment (Brend-Smits) [inline Repositories/ProjectRepository.cs:335]: Refer to my PR #335 please

File: Repositories/ProjectRepository.cs
Comment (Brend-Smits) [inline Repositories/ProjectRepository.cs:332]: FindUserProjectsAsync is sufficient I think

File: Services/Services/ProjectService.cs
Comment (Brend-Smits) [inline Services/Services/ProjectService.cs:155]: FindUserProjectsAsync is sufficient

**Review by Brend-Smits — CHANGES_REQUESTED** [review]

File: API/Controllers/UserController.cs
Comment (niraymak) [inline API/Controllers/UserController.cs:60]: Typo: retreiving -> retrieving

**Review by niraymak — CHANGES_REQUESTED** [review]

**Comment by waltersajtos** [comment]
I still need to fix the merge conflicts for the tests, I will do this once #77 and #354 are merged so we don't need to fix the merge conflicts there

File: Repositories/ProjectRepository.cs
Comment (niraymak) [inline Repositories/ProjectRepository.cs:308]: Collaborators are already included on line 298, right? I think these lines can be removed.

**Review by niraymak — CHANGES_REQUESTED** [review]
Great work! 

I found some redundant code which slows down our application. You can remove those the lines. I already tested it and it works.

File: Repositories/ProjectRepository.cs
Comment (niraymak) [reply Repositories/ProjectRepository.cs:308]: > Great work!
> 
> I found some redundant code which slows down our application. You can remove those the lines. I already tested it and it works.

And don't forget to change the changelog

File: Postman/dex.postman_collection.json
Comment (Ruby77) [inline Postman/dex.postman_collection.json:8600]: Replace copy by an actual name

**Review by Ruby77 — CHANGES_REQUESTED** [review]
Looks good! I'll approve after you resolved Niray's feedback :smile:

File: CHANGELOG.md
Comment (Ruby77) [inline CHANGELOG.md]: Final request change :smile: . I will approve after this change!

```suggestion
- Added functionality to retrieve a user's project - [#313](https://github.com/DigitalExcellence/dex-backend/issues/313)
```

File: CHANGELOG.md
Comment (Ruby77) [inline CHANGELOG.md:43]:
```suggestion
- Added ability to retrieve all the user's projects - [#313](https://github.com/DigitalExcellence/dex-backend/issues/313)
```

**Review by Ruby77 — CHANGES_REQUESTED** [review]

**Review by Ruby77 — APPROVED** [review]
Good work. Everything works as expected!

**Review by niraymak — APPROVED** [review]
💯

**Pull request merged**
